internal/attestation/attestation.go

```diff
@@ -28,11 +30,15 @@ var (
 	AT004 = e.NewError("AT004", "Unsupported attestation predicate type", e.ErrorExitStatus)
 )
 
+type Statement interface {
+	in_toto.ProvenanceStatementSLSA02 | any
```

Comment:
This is only used as the return type of `Attestation.Statement()`. That method is only called from `ApplicationSnapshotImage.WriteInputFile` when building up the large input JSON. I think `any` is probably the right type for that. It just has to be something that can be encoded in JSON.

Reply:
I need to add a method on the `Statement`, the `| any` is here only so that the tests can be a bit less verbose. In fact I was thinking of removing it and keeping a type safe list of supported attestation formats. We want to be careful of the data we pass to the rules and eventually output.

Reply:
IMO the payload of an attestation that has been properly signed is safe to forward to the policies. I don't think we should care too much about the type unless we are doing additional validation in ec-cli.

Reply:
We have `Data` method on `Attestation` that returns the raw bytes, as found on the layer, the `Statement` method should return something that is usable (type safe). I think I’ll remove the `any` from it.

Reply:
Hmm, doesn't compile without `| any`, I'm going to remove the `Statement` type and return `any` from the `Statement` method...